src/Selda/Query/Class.purs

```diff
@@ -9,26 +9,26 @@ import Data.Either (Either, either)
 import Effect.Aff (Aff)
 import Effect.Aff.Class (class MonadAff, liftAff)
 import Foreign (Foreign)
-import Heterogeneous.Folding (class HFoldl, hfoldl)
 import Selda.Col (Col)
 import Selda.Table (Table)
 import Selda.Query.ShowStatement (class GenericShowInsert, genericShowInsert)
 import Selda.Query.Type (FullQuery)
-import Selda.Query.Utils (RecordToArrayForeign(..))
 import Type.Proxy (Proxy)
 
 class GenericQuery ∷ ∀ k. k → (Type → Type) → Row Type → Row Type → Constraint
 class Monad m <= GenericQuery b m i o | i → o, b → m where
   genericQuery
     ∷ Proxy b
     → FullQuery b { | i }
+    → (Array Foreign -> Either String { | o })
     → m (Array { | o })
```

Comment:
With the addition of this new argument, I'm not sure what the order of the arguments should be.